src/spec.c

```diff
@@ -726,13 +726,18 @@ void IndexSpec_Free(void *ctx) {
 }
 
 void IndexSpec_FreeSync(IndexSpec *spec) {
+  //  todo:
+  //  mark I think we only need IndexSpec_FreeInternals, this is called only from the
+  //  LLAPI and there is no need to drop keys cause its out of the key space.
+  //  Let me know what you think
+
```

Comment:
geo... I seem to recall that there was some keyspace interaction. Run the t_llapi test through valgrind, and see if indeed everything is actually deleted.

Reply:
valgrind test failing with it or without it.
I will leave it with a todo to check this out.